pubsub/pulsar/pulsar.go

```diff
@@ -137,11 +139,17 @@ func parsePulsarMetadata(meta pubsub.Metadata) (*pulsarMetadata, error) {
 				value:    v,
 			}
 		} else if strings.HasSuffix(k, topicAvroSchemaIdentifier) {
-			topic := k[:len(k)-len(topicJSONSchemaIdentifier)]
+			topic := k[:len(k)-len(topicAvroSchemaIdentifier)]
```

Comment:
This change correctly fixes `topicJSONSchemaIdentifier` to `topicAvroSchemaIdentifier`, however functionality wise this worked as intended as both are of the same length.